Pipfile

```diff
@@ -4,10 +4,12 @@ url = "https://pypi.org/simple"
 verify_ssl = true
 
 [dev-packages]
-pyrchain = {editable = true,extras = ["dev"],path = "."}
+pyrchain = {path = ".",extras = ["dev"],editable = true}
+mypy-protobuf = "*"
 
 [packages]
-pyrchain = {editable = true,path = "."}
+pyrchain = {path = ".",editable = true}
+typing-extensions = "*"
```

Comment:
@woky Actually I don't understand why there is only `pyrchain` here while there are a bunch of packages in the `Pipfile.lock`. How did you do that?

Reply:
@zsluedem Please put all dependencies in `setup.py`. Since we publish it on PyPI, it makes sense to use `setuptools`. I have little experience with distributing Python packages, but when you create packages (_wheels_), the dependency metadata is generated from `setup.py`.

See also https://github.com/pypa/pipenv/issues/1263#issuecomment-362600555

I don't quite understand the `Pipfile` neither, but it seems that
```
pyrchain = {path = ".",editable = true}
```
picks `install_requires` and
```
pyrchain = {path = ".",extras = ["dev"],editable = true}
```
picks `extra_require['dev']` from `setup.py`. So you can leave `Pipfile` intact and just add it to `setup.py` and it will magically work for both. I.e. when you do `pipenv install` it'll pick everything from `setup.py`.

I think you can move `typing-extensions` also in `extra_require['dev']` because AFAIK Python typing annotations are ignored at runtime.